packages/graphql/src/modules/user/app/authenticateFacebookUser.ts

```diff
@@ -0,0 +1,123 @@
+import StorageProvider from '../../../shared/providers/StorageProvider/models/StorageProvider';
+import {PROVIDERS} from '../../../shared/constants';
+import SocialMediaRepository from '../domain/SocialMediaRepository';
+import User, {ProvidersId} from '../domain/User';
+import UsersRepository from '../domain/UsersRepository';
+import TokenProvider from '../providers/TokenProvider/model/TokenProvider';
+
+interface Dependencies {
+  facebookRepository: SocialMediaRepository;
+  userRepository: UsersRepository;
+  storageProvider: StorageProvider;
+  tokenProvider: TokenProvider;
+}
+
+interface SaveFbDTO {
+  token: string;
+  expires: number;
+  permissions: string[];
+}
+
+export interface SaveFBResponse {
+  user: User;
+  token: string;
+}
+
+const authenticateFacebookUser = ({
```

Comment:
@Beigelman btw shouldn't this be part of a different PR? I mean, the point here is to refactor the backend architecture right?

Reply:
The previous implementation was basically doing everything in one file, since I've isolated different parts of the application I had to refactor this use case as well.